### PR TITLE
added documentation for JOINs

### PR DESCRIPTION
**Pull request by chaudum**
this documentation reflects the current status of JOINs

### CODE REVIEW

File: docs/sql/joins.txt [hunk 1]
```diff
@@ -0,0 +1,166 @@
+.. highlight:: psql
+.. _sql_joins:
+
+Joins
+=====
+
+.. note::
+
+    The JOIN functionality described in this document is currently only
+    available in `nightly builds`_!
```
Comment (dobe) [inline docs/sql/joins.txt]: this can be removed?

File: docs/sql/joins.txt [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+Join Conditions
+...............
+
+Crate supports all operators and scalar functions as join conditions in the
+``WHERE`` clause.
+
+.. note::
+
+    Predicates are **not** supported as join conditions.
```
Comment (mfelsche) [inline docs/sql/joins.txt]: maybe link to ``:ref:`predicates` ``

File: docs/sql/joins.txt [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+However, there are no specific optimisations for certain join conditions such
+as ``=`` (equals) yet. The conditions are implemented as a generic nested loop
```
Comment (mfelsche) [inline docs/sql/joins.txt]: The JOIN operation is implemented as ...

File: docs/sql/joins.txt [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+Roadmap
+-------
+
+*Date: Nov 16th, 2015*
+
+Currently working on
+....................
+
+* Remove the limitation that you have to provide the selected columns in the 
+  ``ORDER BY`` clause when joinind user tables.
```
Comment (mfelsche) [inline docs/sql/joins.txt]: typo: joining

File: docs/sql/joins.txt [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+* Support for joining more than 2 tables.
+
+Upcoming
+........
+
+* Optimize join condition for Equi-Joins.
```
Comment (mfelsche) [inline docs/sql/joins.txt]: condition -> operation

File: docs/sql/queries.txt [hunk 6]
```diff
@@ -48,9 +48,10 @@ Aliases can be used to change the output name of the columns::
 FROM Clause
 -----------
 
-The ``FROM`` clause is used to reference the table to query. A table is referenced
-by schema and table name and can optionally be aliased. If the table ``t`` is
-only referenced by table name, crate assumes the table ``doc.t`` was meant.
+The ``FROM`` clause is used to reference one or two tables.
```
Comment (mfelsche) [inline docs/sql/queries.txt]: ... to reference the relation this select query is based upon. Can be a single table, many tables or a table :ref:`JOIN <sql_joins>`.

File: docs/sql/queries.txt [hunk 7]
```diff
@@ -48,9 +48,10 @@ Aliases can be used to change the output name of the columns::
...
+A table is referenced by schema and table name and can optionally be aliased.
```
Comment (mfelsche) [inline docs/sql/queries.txt]: maybe begin a separate paragraph here.

File: docs/sql/joins.txt [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+.. _sql_joins:
+
+Joins
+=====
+
+Cross Joins
+-----------
+
+Referencing two tables results in a ``CROSS JOIN``.
+The result contains every possible combination (`Cartesian Product`_)
```
Comment (mfelsche) [inline docs/sql/joins.txt]: The result is computed by creating every possible combination ... and then applying the given query operation on it (WHERE clause, SELECT list, ORDER BY clause, ...):

File: docs/sql/reference/select.txt [hunk 9]
```diff
@@ -12,7 +12,7 @@ Synopsis
 ::
 
     SELECT [ ALL | DISTINCT ] * | expression [ [ AS ] output_name ] [, ...]
-      FROM table_ident [ [AS] table_alias ]
+      FROM table_reference [{CROSS JOIN | , } table_reference [{CROSS JOIN | ,} ...]]
```
Comment (mfelsche) [inline docs/sql/reference/select.txt]: pls rewrite FROM synopsis as ``relation``

and split it up in a subrule to: ``table_reference | joined_relation``

where joined_relation:

``relation join_operator relation``

``join_operator`` then is ``, | CROSS JOIN``

this is more readable and future-proof

**Comment by mfelsche** [comment]
Aaaaaaaaaaaaack

**Pull request merged**